programs/cleos/main.cpp

```diff
@@ -466,6 +466,27 @@ fc::variant push_actions(std::vector<chain::action>&& actions, const std::vector
    return push_transaction(trx, signing_keys);
 }
 
+void print_return_value( const fc::variant& at ) {
+   std::string return_value, return_value_prefix{"return value: "};
+   const auto  & iter_value = at.get_object().find("return_value_data");
+   const auto  & iter_hex   = at.get_object().find("return_value_hex_data");
```

Comment:
In normal circumstances, I would agree with you 100%. This time I'm not sure. 
IMHO, formatting the code this way makes it easier to understand. Readability makes up for the extra complexity.
There are my reasons:
- I believe, Kevin or Larry expects 99% of activities to not have return values. In these 99% of cases, the algorithm has to search for `return_value_hex_data` anyway.
- The code is a part of `cleos` only. 'cleos' gets executed by end users. An extra `find()` call which has `O(n)` complexity executed on an `std::vector` of ~20 elements will not slow down the program that much, probably. I believe people who run a bunch of actions at once use `--json` to validate the whole output instead of parsing truncated lines generated by `print_action()`.